app/helpers/application_helper.rb

```diff
@@ -630,6 +631,13 @@ def password_complexity_requirements
     s
   end
 
+  def password_rules_html_format
+    rules = OpenProject::Passwords::Evaluator.active_rules_locales
+    content_tag :ul do
+      rules.map { |item| concat(content_tag(:li, item)) }
+    end
+  end
```

Comment:
:-1: for extending the `application_helper.rb`.

The file should not be bloated anymore than it already is. I would advise to extract the methods you need into your very own helper.

I am also not really sure on why you build the HTML in the Helper, instead of putting it into a reusable partial.